packages/x-date-pickers/src/internals/utils/slots-migration.ts

```diff
@@ -0,0 +1,65 @@
+// TODO v7: This file exist only to simplify typing between
```

Comment:
Nitpick: this file name makes for interesting looking imports. 🙈 
Maybe we could go with something less related to migration? After all, it is going to live in files for the whole v6 lifecycle...
Maybe just `slots-types`? 🤔

Reply:
What is in it is only for migration aspect: capitalizing keys. For v7 we should delete it and make typescript pass. If latter on we need utils that are staying (I don't think we will) we should put them in a distinct file